x-pack/plugins/ml/server/models/job_service/jobs.ts

```diff
@@ -171,6 +172,11 @@ export function jobsProvider(mlClusterClient: ILegacyScopedClusterClient) {
         description: job.description || '',
         groups: Array.isArray(job.groups) ? job.groups.sort() : [],
         processed_record_count: job.data_counts?.processed_record_count,
+        earliestStartTimestampMs: getEarliestDatafeedStartTime(
+          dataCounts?.latest_record_timestamp as number,
```

Comment:
it doesn't look like the `number` type assertions are needed here.
they're also not need for the call to `getLatestDataOrBucketTimestamp`

Reply:
I removed the casting check for both here https://github.com/elastic/kibana/pull/71976/commits/09ae89053a3775799991e9872443a446c3a4086a since they are both not needed.